docs/Getting Started/Slackware/Root on ZFS.rst

```diff
@@ -0,0 +1,232 @@
+
+#. In legacy BIOS mode, using lilo or grub or similar: lilo doesn't understand zfs and even the latest grub understands it with some limitations (for example, no zstd compression). I wasn't able to get grub to work reliably. If you're stuck with legacy BIOS mode, the best option is to put /boot on a separate partition that your loader understands (for example, ext4).
+
+Outside the PC world, the only other platform supported by Slackware that I run is a Raspberry PI: in this case you have a situation similar to UEFI on the PC, where the firmware loads kernel and initrd from a filesystem that it knows about (FAT32 in this case), so again nothing to worry about.
+
+The simplest setup, discussed in this recipe, is the one using UEFI. As said above, if you boot in legacy BIOS mode, you will have to ensure that the boot loader of your choice can load the kernel image.
+
+
+Partition layout
+----------------
+
+Repartitioning an existing system disk in order to make room for a zfs root partition is left as an exercise to the reader (there's nothing specific to zfs).
+
+On my laptop, I resized the ext4 filesystem to half of available space, used sfdisk to relocate into the second half of the disk, inserted a ZFS partition before it, and copied stuff across. This approach has the benefit of providing some kind of recovery mechanism in case stuff goes wrong. When everything is okay, I then deleted the ext4 partition and enlarged the ZFS one.
+
+In any case you will want to have a rescue cdrom at hand, and one that supports zfs out of the box. A Ubuntu live CD will do.
+
+For this recipe, we'll be assuming that we're booting in UEFI mode and there's a single disk configured like this:
+
+.. code-block:: sh
+
+  /dev/sda1 # EFI system partition
+  /dev/sda2 # zfs pool (contains the "root" filesystem)
+
+..
+
+Since we are creating a zpool inside a disk partition (as opposed to using up a whole disk), make sure that the partition type is set correctly (for GPT, 54 or 67 are good choices).
+
+When creating the zfs filesystem, we want to set "mountpoint=legacy" so that the filesystem can be mounted with "mount" in a traditional way; Slackware startup and shutdown scripts expect that.
+
+Back to our recipe, this is a working example:
+
+.. code-block:: sh
+
+  zpool create -o ashift=12 -O mountpoint=none tank /dev/sda2
+  zfs create -o mountpoint=legacy -o compression=zstd tank/root
+  # add more as needed:
+  # zfs create -o mountpoint=legacy [..] tank/home
+  # zfs create -o mountpoint=legacy [..] tank/usr
+  # zfs create -o mountpoint=legacy [..] tank/opt
+
+..
+
+Tweak options to taste; "mountpoint=legacy" is required for the root filesystem, and (at least on Slackware) I like to apply it to all my zfs filesystems.
+
+Also, as a personal preference, I like to set "mountpoint=none" on the pool itself so it's not mounted anywhere by default (do note that zpool's "mountpoint=none" wants an uppercase "-O").
```

Comment:
Let's not use `I` in official docs

Reply:
Hi,
Sorry for the delay due to the holiday season. I have reviewed the documentation per your suggestions.
Feel free to mention my name and/or this document as you see fit.